hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -2160,3 +2160,105 @@ def hpat_pandas_series_median_impl(self, axis=None, skipna=True, level=None, num
         return numpy.median(self._data)
 
     return hpat_pandas_series_median_impl
+
+
+@overload_method(SeriesType, 'argsort')
+def hpat_pandas_series_argsort(self, axis=0, kind='quicksort', order=None):
+    """
+    Pandas Series method :meth:`pandas.Series.argsort` implementation.
+
+    .. only:: developer
+
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_argsort1
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_argsort2
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_argsort_noidx
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_argsort_idx
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_argsort_parallel
+
+    Parameters
+    -----------
+    self: :class:`pandas.Series`
+        input arg
+    axis: :obj:`int`
+        Has no effect but is accepted for compatibility with numpy.
+    kind: {‘mergesort’, ‘quicksort’, ‘heapsort’}, default ‘quicksort’
+        Choice of sorting algorithm. See np.sort for more information. ‘mergesort’ is the only stable algorithm
+    order: None
+        Has no effect but is accepted for compatibility with numpy.
+
+    Returns
+    -------
+    :obj:`pandas.Series`
+         returns: Positions of values within the sort order with -1 indicating nan values.
+    """
+
+    _func_name = 'Method argsort().'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
+
+    if not isinstance(self.data.dtype, types.Number):
+        raise TypingError('{} Currently function supports only numeric values. Given data type: {}'.format(_func_name,
+                                                                                             self.data.dtype))
+
+    if not (isinstance(axis, types.Omitted) or isinstance(axis, types.Integer) or axis == 0):
```

Comment:
Actually `axis` can be represented as `string`, not only `integer`.

Reply:
In pandas' docs axis can be only integer

Reply:
`axis : {0 or ‘index’}, default 0. The value ‘index’ is accepted for compatibility`
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.sort_values.html